File: AndroidRVGooglePlay/app/src/main/java/com/example/shakil/androidrvgoogleplay/Adapter/MyItemAdapter.java
package com.example.shakil.androidrvgoogleplay.Adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.shakil.androidrvgoogleplay.Interface.IItemClickListener;
import com.example.shakil.androidrvgoogleplay.Model.ItemData;
import com.example.shakil.androidrvgoogleplay.R;
import com.squareup.picasso.Picasso;

import java.util.List;

public class MyItemAdapter extends RecyclerView.Adapter<MyItemAdapter.MyViewHolder> {

    private Context context;
    private List<ItemData> itemDataList;

    public MyItemAdapter(Context context, List<ItemData> itemDataList) {
        this.context = context;
        this.itemDataList = itemDataList;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View itemView = LayoutInflater.from(context).inflate(R.layout.layout_item, viewGroup, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, final int i) {
        myViewHolder.txt_item_title.setText(itemDataList.get(i).getName());
        Picasso.get().load(itemDataList.get(i).getImage()).into(myViewHolder.img_item);

        myViewHolder.setiItemClickListener(new IItemClickListener() {
            @Override
            public void onItemClickListener(View view, int position) {
                Toast.makeText(context, "" + itemDataList.get(i).getName(), Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public int getItemCount() {
        return (itemDataList != null ? itemDataList.size() : 0);
    }

    public class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        TextView txt_item_title;
        ImageView img_item;

        IItemClickListener iItemClickListener;

        public void setiItemClickListener(IItemClickListener iItemClickListener) {
            this.iItemClickListener = iItemClickListener;
        }

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            txt_item_title = itemView.findViewById(R.id.tvTitle);
            img_item = itemView.findViewById(R.id.itemImage);

            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            iItemClickListener.onItemClickListener(v, getAdapterPosition());
        }
    }
}

File: KotlinRVGPlay/app/src/main/java/com/example/shakil/kotlinrvgplay/Adapter/MyItemAdapter.kt
package com.example.shakil.kotlinrvgplay.Adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.example.shakil.kotlinrvgplay.Model.ItemData
import android.widget.TextView
import com.example.shakil.kotlinrvgplay.Interface.IItemClickListener
import com.example.shakil.kotlinrvgplay.R
import android.widget.Toast
import com.squareup.picasso.Picasso

class MyItemAdapter(private val context: Context, private val itemList:List<ItemData>?):RecyclerView.Adapter<MyItemAdapter.MyViewHolder>() {
    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): MyViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.layout_item, p0, false)
        return MyViewHolder(view)
    }

    override fun getItemCount(): Int {
        return itemList?.size ?: 0
    }

    override fun onBindViewHolder(p0: MyViewHolder, p1: Int) {
        p0.txt_title.text = itemList!![p1].name
        Picasso.get().load(itemList[p1].image).into(p0.img_item)

        p0.setiItemClickListener(object : IItemClickListener{
            override fun onItemClickListener(view: View, position: Int) {
                Toast.makeText(context, "" + itemList[p1].name, Toast.LENGTH_SHORT).show();
            }

        })
    }

    inner class MyViewHolder(view: View): RecyclerView.ViewHolder(view), View.OnClickListener {

        var txt_title: TextView
        var img_item: ImageView

        lateinit var iItemClickListener: IItemClickListener

        fun setiItemClickListener(iItemClickListener: IItemClickListener) {
            this.iItemClickListener = iItemClickListener
        }

        init {
            txt_title = view.findViewById(R.id.tvTitle) as TextView
            img_item = view.findViewById(R.id.itemImage) as ImageView

            view.setOnClickListener(this)
        }

        override fun onClick(v: View?) {
            iItemClickListener.onItemClickListener(v!!, adapterPosition)
        }

    }
}